packages/web-api/src/controllers/scan-request-controller.ts

```diff
@@ -39,36 +41,56 @@ export class ScanRequestController extends ApiController {
         }
 
         const batchId = this.guidGenerator.createGuid();
-        const response = this.createScanRunBatchResponse(batchId, payload);
+        const processedData = this.getProcessedRequestData(batchId, payload);
 
-        await this.scanDataProvider.writeScanRunBatchRequest(batchId, response);
+        await this.scanDataProvider.writeScanRunBatchRequest(batchId, processedData.requestToBeSaved);
 
         this.context.res = {
             status: 202, // Accepted
-            body: response,
+            body: processedData.responseToBeSent,
         };
 
         this.logger.logInfo('Accepted scan run batch request', {
             batchId: batchId,
-            totalUrls: response.length.toString(),
-            invalidUrls: response.filter(i => i.error !== undefined).length.toString(),
+            totalUrls: processedData.responseToBeSent.length.toString(),
+            invalidUrls: processedData.responseToBeSent.filter(i => i.error !== undefined).length.toString(),
         });
     }
 
-    private createScanRunBatchResponse(batchId: string, scanRunRequests: ScanRunRequest[]): ScanRunResponse[] {
-        return scanRunRequests.map(scanRunRequest => {
+    private getProcessedRequestData(batchId: string, scanRunRequests: ScanRunRequest[]): ProcessedBatchRequestData {
```

Comment:
Function name is little vague, the name of the function should convey what it does.
In this case it determines the scanId(s) for the url(s), or createScantId(s) or generateScanId(s) or assignScanId(s)

Reply:
it is doing more than creating scan id. i named it similar to the return object type